src/lib/HST/Frontend/FromHSE.hs

```diff
@@ -55,11 +53,11 @@ transformDecl (HSE.PatBind s (HSE.PVar _ name) rhs mBinds) = Just
 transformDecl _ = Nothing
 
 -- | Transforms an HSE declaration head into an HST declaration head.
-transformDeclHead :: HSE.DeclHead HSE.SrcSpanInfo -> S.DeclHead HSE
-transformDeclHead (HSE.DHead _ dName    ) = S.DHead (transformName dName)
-transformDeclHead (HSE.DHInfix _ _ dName) = S.DHInfix (transformName dName)
-transformDeclHead (HSE.DHParen _ dHead  ) = S.DHParen (transformDeclHead dHead)
-transformDeclHead (HSE.DHApp _ dHead _  ) = S.DHApp (transformDeclHead dHead)
+transformDeclHead :: HSE.DeclHead HSE.SrcSpanInfo -> S.Name HSE
+transformDeclHead (HSE.DHead _ dName    ) = transformName dName
+transformDeclHead (HSE.DHInfix _ _ dName) = transformName dName
```

Comment:
An error was previously thrown explicitely for infix data declaration heads in the `Application` module. Are those data declarations fully supported now?

Reply:
They are [not supported by the Free Compiler](https://github.com/FreeProving/free-compiler/blob/main/src/lib/FreeC/Frontend/Haskell/Simplifier.hs#L367) at the moment but I don't think that there is a reason not to allow them.

Reply:
See also <https://github.com/FreeProving/free-compiler/pull/139>.